src/tap/src/tapcell.cpp

```diff
@@ -57,17 +63,1091 @@ Tapcell::~Tapcell()
 {
 }
 
-void
-Tapcell::init(Tcl_Interp *tcl_interp,
-	   odb::dbDatabase *db)
+void Tapcell::init(odb::dbDatabase *db, utl::Logger* logger)
 {
   db_ = db;
+  logger_ = logger;
+}
+
+void Tapcell::reset()
+{
+  Tapcell::phy_idx = 0;
+  Tapcell::filled_sites.clear();
+}
+
+//}
+
+//---------------------------------------------------------------
+
+void Tapcell::clear (const char* tap_prefix, const char* endcap_prefix) {
+  int taps_removed = removeCells(tap_prefix);
+  int endcaps_removed = removeCells(endcap_prefix);
+
+
+  // Reset global parameters 
+  Tapcell::phy_idx = 0;
+  Tapcell::filled_sites.clear();
+}
+
+void Tapcell::run(odb::dbMaster* endcap_master, int& halo_x, int& halo_y, const char* cnrcap_nwin_master, const char* cnrcap_nwout_master, const char* endcap_prefix, int& add_boundary_cell, const char* tap_nwintie_master, const char* tap_nwin2_master, const char* tap_nwin3_master, const char* tap_nwouttie_master, const char* tap_nwout2_master, const char* tap_nwout3_master, const char* incnrcap_nwin_master, const char* incnrcap_nwout_master, const char* tapcell_master, int& dist, const char* tap_prefix)
+{
+  cutRows(endcap_master, findBlockages(), halo_x, halo_y);
+  std::vector<std::vector<odb::dbRow*>> rows = organizeRows();
```

Comment:
You have using std::vector above so there is no need for the std:: prefix here.  I would do the same with std::string.